test/unit/deadline_job_attachments/caches/test_caches.py

```
@@ -267,6 +268,251 @@ def test_enter_sqlite_import_error(self, tmpdir):
                 )
                 assert hc.get_entry("/no/file", HashAlgorithm.XXH128) is None
 
+    def test_get_entry_with_byte_range(self, tmpdir):
+        """
+        Tests that a byte range entry is returned when it exists in the cache
+        """
+        # GIVEN
+        cache_dir = tmpdir.mkdir("cache")
+        expected_entry = HashCacheEntry(
+            file_path="large_file.bin",
+            hash_algorithm=HashAlgorithm.XXH128,
+            file_hash="chunk_hash_1",
+            last_modified_time="1234.5678",
+            range_start=0,
+            range_end=268435456,  # 256MB
```

Do we want to cover the case where `range_start > range_end`? Could help clarify whether that should error or be silently accepted.

---

Added a test to ensure this case raises.